src/modules/package.rs

```diff
@@ -367,91 +401,149 @@ java {
         })
         .to_string();
 
-        let expected_version = None;
-        assert_eq!(
-            extract_composer_version(&composer_without_version),
-            expected_version
-        );
+        let project_dir = create_project_dir()?;
+        fill_config(&project_dir, config_name, Some(&config_content))?;
+        expect_output(&project_dir, None, None)?;
+        project_dir.close()
     }
 
     #[test]
-    fn test_extract_project_version() {
-        let project_with_version = toml::toml! {
+    fn test_extract_project_version() -> io::Result<()> {
+        let config_name = "Project.toml";
+        let config_content = toml::toml! {
             name = "starship"
             version = "0.1.0"
         }
         .to_string();
 
-        let expected_version = Some("v0.1.0".to_string());
-        assert_eq!(
-            extract_project_version(&project_with_version),
-            expected_version
-        );
+        let project_dir = create_project_dir()?;
+        fill_config(&project_dir, config_name, Some(&config_content))?;
+        expect_output(&project_dir, Some("v0.1.0"), None)?;
+        project_dir.close()
+    }
 
-        let project_without_version = toml::toml! {
-            [package]
+    #[test]
+    fn test_extract_project_version_without_version() -> io::Result<()> {
+        let config_name = "Project.toml";
+        let config_content = toml::toml! {
             name = "starship"
         }
         .to_string();
 
-        let expected_version = None;
-        assert_eq!(
-            extract_project_version(&project_without_version),
-            expected_version
-        );
+        let project_dir = create_project_dir()?;
+        fill_config(&project_dir, config_name, Some(&config_content))?;
+        expect_output(&project_dir, None, None)?;
+        project_dir.close()
+    }
+
+    fn create_project_dir() -> io::Result<TempDir> {
+        Ok(tempfile::tempdir()?)
+    }
+
+    fn fill_config(
+        project_dir: &TempDir,
+        file_name: &str,
+        contents: Option<&str>,
+    ) -> io::Result<()> {
+        let mut file = File::create(project_dir.path().join(file_name))?;
+        file.write_all(contents.unwrap_or("").as_bytes())?;
+        file.sync_all()
+    }
+
+    fn expect_output(
+        project_dir: &TempDir,
+        contains: Option<&str>,
+        config: Option<toml::Value>,
+    ) -> io::Result<()> {
+        let starship_config = Some(config.unwrap_or(toml::toml! {
+            [package]
+            disabled = false
+        }));
+
+        let actual = render_module("package", project_dir.path(), starship_config);
+        let text = String::from(contains.unwrap_or(""));
+        let expected = Some(format!(
+            "is {} ",
+            Color::Fixed(208).bold().paint(format!("📦 {}", text))
+        ));
+
+        match contains {
+            Some(_contains) => assert_eq!(actual, expected),
+            None => assert_eq!(actual, None),
+        }
```

Comment:
I think it would be better written as the following as you don't need the value `contains`.
```suggestion
        match contains.is_some() {
            true => assert_eq!(actual, expected),
            false => assert_eq!(actual, None),
        }
```

Reply:
changed and updated with upstream

Reply:
after update with upstream, i need to update erlang tests with third parameter in render_module